docs/fsharp/style-guide/component-design-guidelines.md

```diff
@@ -693,11 +693,65 @@ let checkNonNull argName (arg: obj) =
     | null -> nullArg argName
     | _ -> ()
 
-let checkNonNull` argName (arg: obj) =
+let checkNonNull' argName (arg: obj) =
     if isNull arg then nullArg argName
     else ()
 ```
 
+Starting with F# 9, you can leverage the new `| null` [syntax](../language-reference/values/null-values.md#null-values-starting-with-f-9) to make the compiler indicate possible null values and where they need handling. The above code will then start producing warnings:
+
+```fsharp
+let checkNonNull argName (arg: obj) =
+    match arg with
+    // nullness warning: the type 'obj' does not support 'null'
+    | null -> nullArg argName
+    | _ -> ()
+
+let checkNonNull' argName (arg: obj) =
+    // nullness warning: the type 'obj' does not support 'null'
+    if isNull arg then nullArg argName 
+    else ()
+```
+
+To address the warnings, you need to explicitly specify `null` as a possible argument value:
```

Comment:
This set of examples is not very idiomatic, neither the "bad" ones nor the good ones.
The built-in `nullArgCheck` function handles this with two important properties added:
- It does work with type inference and type annotation is not needed
- It uses shadowing to return a non-null value for the binding, so that subsequent code work with a value without nulls.